pkg/console/controllers/clidownloads/controller.go

```diff
@@ -0,0 +1,197 @@
+	v1 "github.com/openshift/api/console/v1"
+	routeclientv1 "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	routesub "github.com/openshift/console-operator/pkg/console/subresource/route"
+	"github.com/openshift/console-operator/pkg/console/subresource/util"
+)
+
+const (
+	controllerWorkQueueKey = "clidownloads-sync-work-queue-key"
+	controllerName         = "ConsoleCliDownloadsSyncController"
+
+	ocCliDownloadsCustomResourceName  = "oc-cli-downloads"
+	odoCliDownloadsCustomResourceName = "odo-cli-downloads"
+
+	LinuxPlatformKey   = "linux"
+	MacPlatformKey     = "mac"
+	WindowsPlatformKey = "windows"
+)
+
+type CliDownloadsSyncController struct {
+	// clients
+	operatorConfigClient      v1helpers.OperatorClient
+	consoleCliDownloadsClient consolev1client.ConsoleCLIDownloadInterface
+	routeClient               routeclientv1.RoutesGetter
+	// listers
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+// factory func needs clients and informers
+func NewCliDownloadsSyncController(
+	operatorClient v1helpers.OperatorClient,
+	consoleClient consolev1client.ConsoleCLIDownloadInterface,
+	routev1Client routeclientv1.RoutesGetter,
+	consoleInformer consolev1informers.ConsoleCLIDownloadInformer,
+	recorder events.Recorder,
+) *CliDownloadsSyncController {
+
+	ctrl := &CliDownloadsSyncController{
+		// clients
+		operatorConfigClient:      operatorClient,
+		consoleCliDownloadsClient: consoleClient,
+		routeClient:               routev1Client,
+		recorder:                  recorder,
+		cachesToSync:              nil,
+		queue:                     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "test"),
+	}
+
+	operatorClient.Informer().AddEventHandler(ctrl.newEventHandler())
+	consoleInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	ctrl.cachesToSync = append(ctrl.cachesToSync,
+		operatorClient.Informer().HasSynced,
+		consoleInformer.Informer().HasSynced,
+	)
+
+	return ctrl
+}
+
+func (c *CliDownloadsSyncController) sync() error {
+	// spec, status, resourcesVerion, err
+	spec, _, _, err := c.operatorConfigClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	// only run if we are managed
+	if !management.IsOperatorManaged(spec.ManagementState) {
+		return nil
+	}
+
+	consoleRoute, err := c.routeClient.Routes(api.TargetNamespace).Get(api.OpenShiftConsoleRouteName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	host := routesub.GetCanonicalHost(consoleRoute)
+	requiredOCCliDownloads := platformBasedCliDownloads(host, "amd64", ocCliDownloadsCustomResourceName)
+
+	_, _, cliDownloadsErr := GetOrCreate(c.consoleCliDownloadsClient, requiredOCCliDownloads)
```

Comment:
Looks good so far, I bet we want to have a `HandleDegraded()` call here, with some potential reasons.

Reply:
unfortunatelly `HandleDegraded()` is a `consoleOperator` struct method, so can't use it here. We need either to refact the `status.go` methods to be able to call on different struct, or have a separate function for the `CLIDownloadsSyncController` struct